test/adaptive_load/metrics_evaluator_test.cc

```diff
@@ -365,5 +395,145 @@ TEST(AnalyzeNighthawkBenchmark, UsesBuiltinMetricsPluginForUnspecifiedPluginName
+  nighthawk::adaptive_load::AdaptiveLoadSessionSpec spec;
+
+  const std::string kMetricName = "good-metric";
+  const double kDeprecatedMetricValue = 123.0;
+
+  FakeMetricsPluginConfig metrics_plugin_config;
+  FakeMetricsPluginConfig::FakeMetric* fake_metric =
+      metrics_plugin_config.mutable_fake_metrics()->Add();
+  fake_metric->set_name(kMetricName);
+  fake_metric->set_value(kDeprecatedMetricValue);
+
+  MetricSpec metric_spec;
+  metric_spec.set_metrics_plugin_name("nighthawk.fake_metrics_plugin");
+  metric_spec.set_metric_name(kMetricName);
+
+  *spec.mutable_informational_metric_specs()->Add() = metric_spec;
+
+  const double kOverrideMetricValue = 1337;
+  nighthawk::client::ExecutionResponse nighthawk_response = MakeNighthawkResponseWithSendRate(1.0);
+  absl::flat_hash_map<std::string, MetricsPluginPtr> name_to_custom_metrics_plugin_map;
+  std::unique_ptr<MockMetricsPluginWithReportingPeriod> plugin =
+      std::make_unique<MockMetricsPluginWithReportingPeriod>(metrics_plugin_config);
+  EXPECT_CALL(*plugin, GetMetricByNameWithReportingPeriod(_, _))
+      .WillRepeatedly(Return(kOverrideMetricValue));
+  name_to_custom_metrics_plugin_map["nighthawk.fake_metrics_plugin"] = std::move(plugin);
+
+  MetricsEvaluatorImpl evaluator;
+  absl::StatusOr<BenchmarkResult> result_or = evaluator.AnalyzeNighthawkBenchmark(
+      nighthawk_response, spec, name_to_custom_metrics_plugin_map);
+  ASSERT_TRUE(result_or.ok());
+  ASSERT_GT(result_or.value().metric_evaluations().size(), 0);
+  EXPECT_EQ(result_or.value().metric_evaluations()[0].metric_value(), kOverrideMetricValue);
+}
+
+TEST(AnalyzeNighthawkBenchmark, FailsWithNoResults) {
+  nighthawk::adaptive_load::AdaptiveLoadSessionSpec spec;
+  nighthawk::client::ExecutionResponse nighthawk_response = MakeNighthawkResponseWithSendRate(1.0);
+  nighthawk_response.mutable_output()->clear_results();
+  absl::flat_hash_map<std::string, MetricsPluginPtr> name_to_custom_metrics_plugin_map;
+
+  MetricsEvaluatorImpl evaluator;
+  absl::StatusOr<BenchmarkResult> result_or = evaluator.AnalyzeNighthawkBenchmark(
+      nighthawk_response, spec, name_to_custom_metrics_plugin_map);
+  ASSERT_FALSE(result_or.ok());
+  EXPECT_EQ(result_or.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_THAT(result_or.status().message(), HasSubstr("output.results cannot be empty."));
+}
+
+// Utility to set the execution time and execution duration of a results object.
+void SetTimeAndDurationOnResult(nighthawk::client::Result* result, int64_t time_seconds,
+                                int64_t duration_seconds) {
+  result->mutable_execution_start()->set_seconds(time_seconds);
+  result->mutable_execution_start()->set_nanos(0);
+  result->mutable_execution_duration()->set_seconds(duration_seconds);
+  result->mutable_execution_duration()->set_nanos(0);
+}
+
+TEST(AnalyzeNighthawkBenchmark, FailsWithDurationWhereAllWorkersAreActive) {
```

Comment:
Should the name of the test case be in the negative, e.g. `FailsWithDurationWhereAllWorkersAreNeverActive` or `FailsWithoutDurationWhereAllWorkersAreActive`.

Reply:
Done.